pandas/core/categorical.py

```diff
@@ -882,6 +882,7 @@ def remove_unused_categories(self, inplace=False):
 
         if idx.size != 0 and idx[0] == -1:  # na sentinel
             idx, inv = idx[1:], inv - 1
+        inv = inv.astype(np.int8)
```

Comment:
we could add an arg to ``pd.unique()`` to return the inverses, which we have (i think directly)